android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SerpTelemetryRepository.kt

```diff
@@ -26,7 +26,7 @@ class SerpTelemetryRepository(
     rootStorageDirectory: File,
     private val readJson: () -> JSONObject,
     collectionName: String,
-    serverUrl: String = REMOTE_ENDPOINT_URL,
+    serverUrl: () -> String = { REMOTE_PROD_ENDPOINT_URL },
```

Comment:
Since this parameter is a lambda, can we update the name to communicate that it's doing something? `getServerUrl` for example.

Rather than pass the `serverUrl` to the `RemoteSettingsClient` in this class, we should rebuild the client every time we go to use it so that it can change during runtime. I would suggest a small factory function for this. Then, your use cases would look something like:

```
fun buildClient() = RemoteSettingsClient(getServerUrl(), bucketName, collectionName, storageRootDirectory)

internal suspend fun writeToCache(records: RemoteSettingsResponse): RemoteSettingsResult {
    return buildClient().write(records)
}
```

Reply:
resolving based on Matt's below comment